### PR TITLE
Remove padding using new encryption API

### PR DESCRIPTION
**Pull request by lumag**
This is a reimplementation of RSA encryption/decryption API on top of #245 .

cc @tarcieri @newpavlov @dignifiedquire

### CODE REVIEW

**Comment by lumag** [comment]
This PR incorporates #245  instead of replacing it to let us discuss these two PRs separately.

File: src/lib.rs [hunk 1]
```diff
@@ -13,7 +13,9 @@
 //!
 //! ## PKCS#1 v1.5 encryption
 //! ```
-//! use rsa::{PublicKey, RsaPrivateKey, RsaPublicKey, PaddingScheme};
+//! use rsa::{RsaPrivateKey, RsaPublicKey};
+//! use rsa::pkcs1v15::{DecryptingKey, EncryptingKey};
```
Comment (dignifiedquire) [inline src/lib.rs:17]: `EncryptingKey` reads quite odd, I would have expected it to be called `EncryptionKey`

Comment (lumag) [reply src/lib.rs:17]: Ack

**Comment by tarcieri** [comment]
I do like this approach overall.

I worry there's going to be a lot of complaints and confusion for anyone attempting to upgrade who just wants to be able to call an inherent method on a key type.

If we do go with this I think it needs some documentation in the changelog or elsewhere about how to upgrade from the previous API.

**Comment by tarcieri** [comment]
Re: this vs #244, I think both would be interesting?

I think there are plenty of use cases for #244 where a given key might be used for any application with any padding mode, especially KMS/TPM-style applications where a given client might ask for a key operation involving any of them.

However, if you know ahead of time what the key's usage will be, the API in this PR is great.

**Comment by tarcieri** [comment]
@lumag if you can rebase, the encryptor/decryptor types seem interesting

**Comment by lumag** [comment]
Closing in favour of #259

**Pull request closed without merging**